topic-operator/src/main/java/io/strimzi/operator/topic/TopicOperator.java

```diff
@@ -716,7 +716,7 @@ public void onMaxAttemptsExceeded(MaxAttemptsExceededException e) {
 
     /** Called when a resource is added in k8s */
     void onResourceAdded(KafkaTopic addedTopic, Handler<AsyncResult<Void>> resultHandler) {
-        if (resourcePredicate.test(addedTopic)) {
+        if (resourcePredicate.test(addedTopic) || addedTopic.getMetadata().getLabels() == null || addedTopic.getMetadata().getLabels().size() == 0) {
```

Comment:
I would factor out a method getting an `HasMetadata` parameter as input for checking the labels collection.